samples/apps/copilot-chat-app/WebApp/src/libs/useChat.ts

```diff
@@ -25,54 +31,151 @@ export const useChat = () => {
         '/assets/bot-icon-5.png',
     ];
 
-    const getAudienceMemberForId = (id: string, chatId: string, audience: ChatUser[]) => {
-        if (id === `${chatId}-bot` || id.toLocaleLowerCase() === 'bot') return Constants.bot.profile;
+    const loggedInUser: ChatUser = {
+        id: account!.homeAccountId,
+        fullName: account!.name!,
+        emailAddress: account!.username,
+        photo: undefined, // TODO: Make call to Graph /me endpoint to load photo
+        online: true,
+        lastTypingTimestamp: 0,
+    };
+
+    // TODO: Verify consistent 'sender' field -- is it name or ID?
+    const getAudienceMemberForId = (id: string, botId: string, audience: ChatUser[]) => {
+        if (id === `${botId}-bot` || id === 'Bot') return Constants.bot.profile;
         return audience.find((member) => member.id === id);
     };
 
-    // TODO: handle error case of missing account information
+    const getVariableValue = (variables: Variables, key: string): string | undefined => {
+        for (const idx in variables) {
+            if (variables[idx].key === key) {
+                return variables[idx].value;
+            }
+        }
+        // End of array, did not find expected variable.
+        throw new Error(`Could not find valid ${key} variable in context.`);
+    };
+
     const createChat = async () => {
-        const name = `SK Chatbot @ ${new Date().toLocaleString()}`;
-        const user: ChatUser = {
-            id: account?.homeAccountId ?? '',
-            fullName: account?.name ?? 'Unknown User',
-            emailAddress: account?.username ?? '',
-            photo: undefined,
-            online: true,
-            lastTypingTimestamp: 0,
-        };
+        const chatTitle = `SK Chatbot @ ${new Date().toLocaleString()}`;
+        try {
+            var ask: IAsk = {
+                input: chatTitle,
+                variables: [
+                    { key: 'userId', value: account!.homeAccountId! },
+                    { key: 'userName', value: account!.name! },
+                ],
+            };
 
-        const botProfilePictureIndex = Object.keys(conversations).length % botProfilePictures.length;
-        const newChat: ChatState = {
-            id: name,
-            messages: [initialBotMessage(account?.name ?? 'there')],
-            audience: [user],
-            botTypingTimestamp: 0,
-            botProfilePicture: botProfilePictures.at(botProfilePictureIndex) ?? '/assets/bot-icon-1.png',
-        };
-        dispatch(addConversation(newChat));
-        dispatch(setSelectedConversation(name));
-        return name;
+            await sk.invokeAsync(ask, 'ChatMemorySkill', 'CreateChat').then(async (result: IAskResult) => {
+                const newChatId = result.value;
+                const initialBotMessage = getVariableValue(result.variables, 'initialBotMessage');
+
+                // TODO: hook up chat object once Tao makes change to return Chat object
```

Comment:
Is this still required?

Reply:
I think what we have now is good. Will remove the TODO.